### PR TITLE
NAS-114960 / 22.12 / Report true S.M.A.R.T. self-test progress

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
time 6:00

File: src/middlewared/middlewared/plugins/smart.py [hunk 1]
```diff
@@ -19,17 +19,20 @@
...
     if disk["disk"] is None:
         return
 
-    args = await get_smartctl_args(middleware, devices, disk["disk"], disk["smartoptions"])
-    if args:
-        p = await smartctl(args + ["-l", "selftest"], check=False, encoding="utf8")
-        tests = parse_smart_selftest_results(p.stdout)
-        if tests is not None:
-            return dict(tests=tests, **disk)
+    output = await middleware.call("disk.smartctl", disk["disk"], ["-a"], {"silent": True})
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/smart.py:29]: Why not use `{"cache": True}`?

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/smart.py:29]: Oops, nvm by default it's set to True 😄

**Comment by themylogin** [comment]
backport

**Pull request merged**